onnx/test/shape_inference_test.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from onnx import checker, helper, ModelProto, TensorProto
+
+import onnx.shape_inference
+import unittest
+
+class TestShapeInference(unittest.TestCase):
+
+    def _inferred(self, graph):
+        orig_model = helper.make_model(graph, producer_name='onnx-test')
+        orig_model_str = orig_model.SerializeToString()
+        inferred_model_str = onnx.shape_inference.infer_shapes(orig_model_str)
+        inferred_model = ModelProto()
+        inferred_model.ParseFromString(inferred_model_str)
+        checker.check_model(inferred_model)
+        return inferred_model
+
+    def test_transpose(self):
```

Comment:
more tests? e.g. for corner cases when value_info is already populated with something

Reply:
it should be fairly easy to hook it up with the backend tests (see my comment above re. `DummyBackend`)

Reply:
++